nntrainer/optimizers/optimizer_devel.h

```diff
@@ -125,6 +125,14 @@ class Optimizer : public ml::train::Optimizer {
    */
   virtual void checkValidation() const;
 
+  /**
+   * @brief     Add extra variables per weight if the optimizer needs any.
+   * @param dim Dimension of tensor to be added as a optimizer variable
+   * @retval #ML_ERROR_NONE Successful.
+   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
+   */
+  virtual std::vector<TensorDim> addOptimizerVariable(TensorDim dim) = 0;
```

Comment:
please consider the signature `std::vector<TensorDim> addOptimizerVariable(const TensorDim &dim)`

Reply:
Done